gradio/components.py

```diff
@@ -3323,15 +3334,28 @@ def deserialize(
             return None
         gallery_path = os.path.join(save_dir, str(uuid.uuid4()))
         for img_data in x:
-            ImgSerializable.deserialize(self, img_data, gallery_path)
+            if isinstance(img_data, list) or isinstance(img_data, tuple):
+                img_data, caption = img_data
+                prefix = f"[{utils.make_valid_filename(caption)}]-"
```

Comment:
I don't love this approach. It's a bit flaky and the user loses any ability to input special characters in their caption. I would instead create a `captions.json` in the same folder as the images, and read/write from their.

Reply:
You're right, I'll store captions in a json. But I'll preserve the captions in the filename too, I think it makes it easier to view

Reply:
Done.